src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionSettings.cs

```diff
@@ -142,32 +132,6 @@ private static bool AllowHttp2
             }
         }
 
-        private static bool AllowUnencryptedHttp2
-        {
-            get
-            {
-                // Default to not allowing unencrypted HTTP/2, but enable that to be overridden
-                // by an AppContext switch, or by an environment variable being to to true/1.
-
-                // First check for the AppContext switch, giving it priority over the environment variable.
-                if (AppContext.TryGetSwitch(Http2UnencryptedSupportAppCtxSettingName, out bool allowHttp2))
```

Comment:
Would it be possible to leave the AppContext/environmental option in .NET 5 and remove it in 6? That would allow some time for libraries to transition to using HTTP version selection.

I think there will be people who are relying on the switch today (.NET Core 3.x) who will break on upgrading to .NET 5. For people using gRPC then the fix for them will be to upgrade to a newer version of Grpc.Net.Client, but I don't know if that requiring both those things to happen at once is ideal.

If you would prefer that the switch is removed in .NET 5 then I'm also ok with that. I don't feel strongly about this.

Reply:
Sadly, it is non-trivial to maintain both.
Given that the switch is just experimental, it is safe to assume people who depend on it will be willing to update code when migrating to .NET 5.